src/ml_tooling/transformers/functransform.py

```diff
@@ -66,5 +66,8 @@ def _validate_strategy(self, strategy):
 
     def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         x_ = X.copy()
-        x_ = x_.apply(self.func, axis=1).to_frame()
+        if self._func_map.get(self.strategy):
+            x_ = self._func_map.get(self.strategy)(x_, axis=1).to_frame()
```

Comment:
What happens here if we pass multiple columns?

Reply:
I'm not sure i follow. We pass a dataframe. Won't that always have multiple columns?

Reply:
I'm wondering if there is a scenario where `to_frame` will fail because the return value of the apply or func_map strategy returns a DataFrame and not a Series?

Reply:
Well, we are always applying the function vertically in the dataframe (axis=1) so based on that, it should only be possible to return one column - unless of course you are trying to do multiple things at once in a custom function. But then I'd argue that you should write to separate functions.

But we can wrap it in somthing like this to be sure?:

if isinstance(x_, pd.Series):
    x_.to_frame()

Reply:
Either that or we specify in the docstring that the callable should return a Series

Reply:
Sure, I'll add it to the docstring no problem :+1:

Reply:
Done